riscv-osps.adoc

```diff
@@ -0,0 +1,302 @@
+as _general purpose computing_.  For platforms of this type, it is expected
+that the user will be able to add or remove cards such as PCIe or SATA, 
+add or remove devices such as disks, monitors or graphics cards, and will
+be able to choose from one or more available operating systems.  Commercial
+desktop, laptop or server systems are primary examples.
+
+Operating systems for this use case include at a minimum the members
+of the Linux families (Fedora, Debian, Red Hat, SUSE, and similar).  If
+a platform is compliant with the requirements of this use case, it should
+be possible to easily install and maintain any of the Linux distributions
+without modification.
+
+It should also be possible to use any of the other Unix-based operating
+systems and distributions such as those in the BSD family, as well as
+commercial operating systems such as Microsoft Windows.
+
+The stress here is on the idea of _general_ computing, and hence most of
+the readily available operating systems designed for that environment.  The
+goal is to be able to seamlessly replace existing systems based on other
+ISAs.
+
+#### Revision History
+General Purpose Computing Current Revision: *0* [September 2020]
+
+#### RISC-V ISA Requirements
+##### Machine Mode
+##### Hypervisor Mode
+##### Supervisor Mode
+. Supervisor-mode environments must implement at least version 20190608
+  of the RISC-V Instruction Set Manual Volume II: Privileged
+  Specification <<ISAU>>.
+
+. Supervisor-mode environments must implement RV64GC.
+
+. Supervisor-mode environments must implement the Sv39 page-based
+  virtual-memory scheme.   Systems that support Sv48 must support
+  Sv39, systems that support Sv57 must support Sv48, and so forth.
+
+. Unless otherwise specified by a given I/O device, I/O regions are at
+  least point-to-point strongly ordered.  All devices attached to a
+  given PCIe root complex are on the same ordered channel (numbered 2
+  or above), though different root complexes might not be on the same
+  ordering channel.
+
+. On RV64I-based Unix-class systems the negative virtual addresses are
+  reserved for the kernel.
+
+. External devices (DMA engines, the debug unit, non RISC-V cores, etc.)
+  that are visible to RISC-V harts must appear as coherent agents, just
+  like any RISC-V hart would.  If additional ordering constraints are
+  necessary for a device to function, those will be provided by a
+  device-specific mechanism.
+
+##### User Mode
+. User-mode environments must implement at least version 20191213
+  of the RISC-V Instruction Set Manual Volume I: Unprivileged
+  Specification <<ISAP>>.
+
+. User-mode programs may not execute the `fence.i` instruction.
+
+. User-mode environments may provide additional ISA extensions, but if those
+  extensions add user-visible state they must be initially disabled.
+
+. Within main-memory regions, aligned instruction fetch must be atomic,
+  up to the smaller of ILEN and XLEN bits.  In particular, if an aligned
```

Comment:
Don't we require that XLEN=64 bits?

Reply:
Yes, we do.  Thanks.  Should not have cut'n'pasted so glibly.